### PR TITLE
CE-1549 ExactTarget Fix edits updates maintenance script

### PR DESCRIPTION
**Pull request by kamilkoterba**
https://wikia-inc.atlassian.net/browse/CE-1549

Maintenance script for updating number of user edits per wiki in ExactTarget was failing.

There's a limit in number of objects that can be sent in one request. Because update parameters passed to task are nested [ [ user_id ][ wiki_id ] => editsnumber ] slicing of parameters to acceptable chunks were wrong.

Changed a way of preparing parameters by firstly preparing flat array, then slicing into chunks and finally preparing proper nested params structure required by task.

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php [hunk 1]
```diff
@@ -75,6 +85,43 @@ private function getLastDayDate() {
...
+			$aUnmergedKeysAssoc = $this->unmergeKey( $sMergedKey );
+			$iUserId = $aUnmergedKeysAssoc['user_id'];
+			$iWikiId = $aUnmergedKeysAssoc['wiki_id'];
+			$aUsersEditsData[$iUserId][$iWikiId] = $iEditcount;
+		}
+		return $aUsersEditsData;
+	}
+
+	private function mergeKey( $iUserId, $iWikiId ) {
+		return $iUserId . '@' . $iWikiId;
```
Comment (adamkarminski) [inline extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php]: What about using ```implode()``` here?

Comment (kamilkoterba) [inline extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php]: I think it's cheaper to glue it by combining strings in this simple case. Making array of two elements just for using implode is overhead in my opinion.

File: extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php [hunk 2]
```diff
@@ -49,19 +55,23 @@ private function getUserEdits( DatabaseBase $oStatsDBr, $sStartDate ) {
 			->AND_( 'user_id' )->NOT_EQUAL_TO( 0 )
 			->GROUP_BY( 'wiki_id' )
 			->GROUP_BY( 'user_id' );
+
 		$aUsersEditsData = $oWikiaSQL->runLoop( $oStatsDBr, function( &$oUserResult, $oUserEditCountWikiResult ) {
```
Comment (adamkarminski) [inline extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php]: Please change ```$oUserResult``` to ```$aUsersEditsData```.

File: extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php (hunk 1 above)
Comment (adamkarminski) [inline extensions/wikia/ExactTargetUpdates/maintenance/ExactTargetUpdateUserEditsPerWikiMaintenance.php]: Although you're not concatenating strings but integers :) Anyway, it's your call, I won't mind leaving it.

**Pull request merged**